examples/java/src/main/java/org/apache/beam/examples/cookbook/TriggerExample.java

```diff
@@ -393,12 +393,12 @@ public void processElement(ProcessContext c, BoundedWindow window) throws Except
   static class ExtractFlowInfo extends DoFn<String, KV<String, Integer>> {
     @ProcessElement
     public void processElement(ProcessContext c) throws Exception {
-      String[] laneInfo = c.element().split(",");
+      String[] laneInfo = c.element().split(",", -1);
       if ("timestamp".equals(laneInfo[0])) {
         // Header row
         return;
       }
-      if (laneInfo.length < 48) {
+      if (laneInfo.length < 50) {
```

Comment:
Can you elaborate on what this change is for? I'm not sure what this magic number represents. If you are familiar, perhaps extract it out as a named constant that provides some context. i.e. `const int NUM_FIELDS = 50;`

Reply:
This value change is necessary after changing the `c.element().split()` to have limit of `-1`. In the [second input data](https://github.com/apache/beam/blob/master/examples/java/src/test/java/org/apache/beam/examples/cookbook/TriggerExampleTest.java#L60-L61), when we didn't have the limit set, `split()` would actually truncate the fields following after the last `0` on line 61 resulting in length of 45. After setting the limit, it increased the number of fields to 49 for that input. We still maintained 50 fields for the input that starts on line 59, which [textExtractTotalFlow()](https://github.com/apache/beam/blob/master/examples/java/src/test/java/org/apache/beam/examples/cookbook/TriggerExampleTest.java#L100) seems to care about.

Great suggestion on the constant, I'll push up the change with that now.

Reply:
Got it, thanks for the explanation!